ntbs-integration-tests/ServiceDirectory/ServiceDirectorySearchTest.cs

```diff
@@ -234,16 +235,51 @@ private static int GetNumberOfDisplayedResults(IHtmlDocument pageContent)
         private async Task<int> GetTotalExpectedResults(string searchKeyword)
         {
             var searchKeywords = new List<string> {searchKeyword};
+            var context = _factory.Services.GetService<NtbsContext>()!;
             
-            var referenceDataRepository = _factory.Services.GetService<IReferenceDataRepository>()!;
-            var userSearchService = _factory.Services.GetService<IUserSearchService>()!;
-            
-            var expectedRegionResults = (await referenceDataRepository.GetPhecsBySearchKeywords(searchKeywords)).Count;
-            var expectedServiceResults = (await referenceDataRepository.GetActiveTBServicesBySearchKeywords(searchKeywords)).Count;
-            var expectedHospitalResults = (await referenceDataRepository.GetActiveHospitalsBySearchKeywords(searchKeywords)).Count;
-            var expectedUserResults = (await userSearchService.OrderQueryableAsync(searchKeywords)).Count;
+            var expectedRegionResults = await context.PHEC
+                .CountAsync(x => x.Name.ToLower().Contains(searchKeyword));
+            var expectedServiceResults = await context.TbService
+                .Include(t => t.PHEC)
+                .CountAsync(t => t.Name.ToLower().Contains(searchKeyword) && !t.IsLegacy);
+            var expectedHospitalResults = await context.Hospital
+                .Include(h => h.TBService.PHEC)
+                .CountAsync(h => h.Name.ToLower().Contains(searchKeyword) && !h.IsLegacy);
+            var expectedUserResults = await GetExpectedUserResults(context, searchKeywords);
 
             return expectedRegionResults + expectedServiceResults + expectedHospitalResults + expectedUserResults;
         }
+
+        private static async Task<int> GetExpectedUserResults(NtbsContext context, List<string> searchKeywords)
```

Comment:
I think this method is really overkill. We're only calling it once, for a particular search (the search with a single keyword `"a"`) and that's for a good reason - it's a very general search where it's not practical to list out/count the expected results explicitly. We wouldn't want to use this for any of our other tests.

With this in mind, it would be better to explicitly use the keyword `"a"`, or at the very least pass in a single keyword.